service/src/main/java/com/codeforcommunity/processor/authenticated/ProtectedSchoolProcessorImpl.java

```diff
@@ -473,6 +476,102 @@ public ReportWithLibrary createReportWithLibrary(
+      ReportWithoutLibrary temp =
+          db.select(
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.ID,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.CREATED_AT,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.UPDATED_AT,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.SCHOOL_ID,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.USER_ID,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.NUMBER_OF_CHILDREN,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.NUMBER_OF_BOOKS,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.MOST_RECENT_SHIPMENT_YEAR,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.WANTS_LIBRARY,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.HAS_SPACE,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.CURRENT_STATUS,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.REASON_WHY_NOT,
+                  SCHOOL_REPORTS_WITHOUT_LIBRARIES.READY_TIMELINE)
+              .from(SCHOOL_REPORTS_WITHOUT_LIBRARIES)
+              .where(SCHOOL_REPORTS_WITHOUT_LIBRARIES.DELETED_AT.isNull())
+              .and(SCHOOL_REPORTS_WITHOUT_LIBRARIES.SCHOOL_ID.eq(schoolId))
+              .fetchOneInto(ReportWithoutLibrary.class);
+      if (temp == null) {
+        logger.error("Report was not found in table");
+        throw new NoReportFoundException(schoolId);
+      }
+    }
+    if (libraryStatus == LibraryStatus.IN_PROGRESS) {
+      ReportWithLibraryInProgress temp =
+          db.select(
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.ID,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.CREATED_AT,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.UPDATED_AT,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.SCHOOL_ID,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.USER_ID,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.NUMBER_OF_CHILDREN,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.NUMBER_OF_BOOKS,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.MOST_RECENT_SHIPMENT_YEAR,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.IS_SHARED_SPACE,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.HAS_INVITING_SPACE,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.ASSIGNED_PERSON_ROLE,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.ASSIGNED_PERSON_TITLE,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.APPRENTICESHIP_PROGRAM,
+                  SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.TRAINS_AND_MENTORS_APPRENTICES)
+              .from(SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES)
+              .where(SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.DELETED_AT.isNull())
+              .and(SCHOOL_REPORTS_IN_PROGRESS_LIBRARIES.SCHOOL_ID.eq(schoolId))
+              .fetchOneInto(ReportWithLibraryInProgress.class);
+      if (temp == null) {
+        logger.error("Report was not found in table");
+        throw new NoReportFoundException(schoolId);
+      }
+    }
+    throw new NoReportFoundException(schoolId);
```

Comment:
Are we ever returning the report here?

I think it would be less repetitive if we assigned the reports to a variable outside of these `if` statements (you could use the `ReportGeneric` here), and instead of repeating the `if (temp == null)` inside each `if` statement, we can move the check here at the bottom of the function (throw the exception if it's null, or return the report if it's not)